test/hack/checkLabels/main.go

```diff
@@ -33,25 +41,34 @@ func parseInhibitionFile(fileName string) (alertConfig.Config, error) {
 	return inhibitions, nil
 }
 
-// Return the list of target and source labels from the alertmanager config file
-func getTargets(config alertConfig.Config) ([]string, []string) {
-	var targetMatchers []string
-	var sourceMatchers []string
+// Return either the list of target or a specific source label from the alertmanager config file
+func getTargetsAndSources(config alertConfig.Config, target string) ([]string, string) {
+	var am_targetMatchers []string
+	var am_sourceMatchers []string
 
 	for _, match := range config.InhibitRules {
-		for _, target := range match.TargetMatchers {
-			targetMatchers = addIfNotPresent(targetMatchers, target.Name)
-		}
-
-		for _, source := range match.SourceMatchers {
-			// Checking if the label value is a boolean (can't use TypeOf as value is interpreted as string)
-			if source.Value == "true" || source.Value == "false" {
-				sourceMatchers = addIfNotPresent(sourceMatchers, source.Name)
+		for _, targetLabel := range match.TargetMatchers {
+			if target == "" {
+				am_targetMatchers = addIfNotPresent(am_targetMatchers, targetLabel.Name)
+			} else if targetLabel.Name == target {
+				for _, source := range match.SourceMatchers {
+					if source.Value == "true" || source.Value == "false" {
+						am_sourceMatchers = append(am_sourceMatchers, source.Name)
+					}
+				}
+			} else {
+				break
```

Comment:
What is this break for? This lack a comment at least.

Reply:
This is to break out of the current loop if the current label does not match the searched one

Reply:
Removed break.